### ISSUE TITLE
Question 

### ISSUE THREAD

**Issue by surya-narayanan** [tatp22/multidim-positional-encoding#42]
Hi, 

More of a question than a bug - if I have

```
p_enc_2d = PositionalEncoding2D(5)
y = torch.zeros((1,6,2,10))

```

is it not true that 

`p_enc_2d(y)`

should error because p_enc_2d is going to produce an encoding with dim 5, but the data has 10 channels. But it looks like it doesn't produce an error, and instead produces a tensor of shape torch.Size([1, 6, 2, 8]).
